sdk/storage/storage-queue/src/MessagesClient.ts

```diff
@@ -180,12 +184,16 @@ export class MessagesClient extends StorageClient {
    *                     "https://myaccount.queue.core.windows.net/myqueue/messages". You can
    *                     append a SAS if using AnonymousCredential, such as
    *                     "https://myaccount.queue.core.windows.net/myqueue/messages?sasString".
-   * @param {Credential} credential Such as AnonymousCredential, SharedKeyCredential or TokenCredential.
+   * @param {Credential | CoreHttpTokenCredential } credential Such as AnonymousCredential, SharedKeyCredential or TokenCredential.
```

Comment:
There are `corehttp.TokenCredential` and `CoreHttpTokenCredential`. Better to be consistent.

Reply:
I agree, I was also thinking that while making this change.  I might standardize on `CoreHttp*` instead of the `coreHttp.` approach because it occurs more often in the changes I made.

Reply:
I'm worried about the renaming to `CoreHttpTokenCredential`.

Renaming is private in storage SDK. Customers won't know anything about `CoreHttpTokenCredential`. He only can get `TokenCredential` from identity or core-http packages.

Reply:
How about this?

1. Rename Storage SDK's `TokenCredential` class to another name like `RawTokenCredential` or `SimpleTokenCredential`.
 Just some other name different with interface from core-http `TokenCredential`. This is the root conflict we need to resolve.

2. Update Storage SDK's `RawTokenCredential` class to implement `TokenCredential` interface provided by core-http.

3. In constructors of `QueueClient` and other Clients, make credential parameter to accept `Credential | TokenCredential`

```javascript
  constructor(
    url: string,
    credential?: Credential | TokenCredential,
    options?: NewPipelineOptions
  );
```

4. Update all documents and tell customers he can get a `TokenCredential` using identity package or `RawTokenCredential` provided in storage SDK.

In this way, we don't need the `CoreHttpTokenCredential` renaming. No confusing, and all TokenCredential will be aligned to have same definition.

Reply:
I've renamed `TokenCredential` to `RawTokenCredential` as requested and updated references of `CoreHttpTokenCredential` to `TokenCredential`.

Regarding the union of `Credential | TokenCredential`,  I don't know if there's a way around that just yet.  Currently `@azure/core-http`'s `TokenCredential` interface is meant for authentication against Azure Active Directory and using the resulting Bearer token for authentication.  If we expand to other credential types in the future it's possible that we could deprecate `Credential` in favor of `TokenCredential`, but I'm not sure whether that is ideal.